src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+                originalDeclarationNode.Identifier.WithTrailingTrivia(SyntaxFactory.ElasticMarker),
+                originalDeclarationNode.TypeParameterList?.WithTrailingTrivia(SyntaxFactory.ElasticMarker),
+                SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(propertiesToAddAsParams)),
+                // TODO: inheritance in the case where we inherit from a record (because we previously activated on one)
+                originalDeclarationNode.BaseList,
+                originalDeclarationNode.ConstraintClauses,
+                originalDeclarationNode.OpenBraceToken,
+                SyntaxFactory.List(membersToKeep),
+                originalDeclarationNode.CloseBraceToken,
+                semicolonToken: default)
+                .NormalizeWhitespace()
+                .WithLeadingTrivia(modifiedClassTrivia)
+                .WithAdditionalAnnotations(Formatter.Annotation);
+
+            // if we have no members, use semicolon instead
+            if (membersToKeep.IsEmpty)
+            {
+                changedTypeDeclaration = changedTypeDeclaration
+                    .WithOpenBraceToken(default)
+                    .WithCloseBraceToken(default)
+                    .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.SemicolonToken));
+            }
+
+            var changedDocument = await document.ReplaceNodeAsync(
+                originalDeclarationNode, changedTypeDeclaration, cancellationToken).ConfigureAwait(false);
+
+            return changedDocument;
+        }
+
+        private static bool ShouldConvertProperty(MemberDeclarationSyntax member, INamedTypeSymbol containingType)
+        {
+            if (member is not PropertyDeclarationSyntax property)
+            {
+                return false;
+            }
+
+            if (property.Initializer != null || property.ExpressionBody != null)
+            {
+                return false;
+            }
+
+            var propAccessibility = CSharpAccessibilityFacts.Instance.GetAccessibility(member);
+
+            // more restrictive than internal (protected, private, private protected, or unspecified (private by default))
+            if (propAccessibility < Accessibility.Internal)
+            {
+                return false;
+            }
+
+            // no accessors declared
+            if (property.AccessorList == null)
+            {
+                return false;
+            }
+
+            // for class records and readonly struct records, properties should be get; init; only
+            // for non-readonly structs, they should be get; set;
+            // this is bacause those are the defaults for properties generated from positional constructors
+            // for each respective class type, although we could decide to change this requirement
+            // the get should be public
+            // neither should have bodies (as it indicates more complex functionality)
```

Comment:
absense of punctuation/numbering made this really confusing at first :)

Reply:
nit: This is all a refactoring, so i think we should just relax all of this.  but that can be in followup.